src/Bootstrap.h

```diff
@@ -19,6 +20,9 @@ class Bootstrap {
 
 	void showErrorMessage();
 
+	const std::vector<std::wstring> getMinecraftJavaPaths(const HostArchitecture::Value& hostArch);
+
 private:
-	std::shared_ptr<ISystemHelper> systemHelper;
+	ISystemHelper& systemHelper;
```

Comment:
```suggestion
	const ISystemHelper& systemHelper;
```

Reply:
Thanks for taking a look at this :) I have fixed the other points.

I was told a while back that you generally shouldn't have const refrences as member vars, ill be honest I didn't fully understand the reasoning. There seems to be a bit of litrature about this online such as: https://lesleylai.info/en/const-and-reference-member-variables/ And then other places seem to suggest having const is fine. 

The orignal design around this is to allow a mock of ISystemHelper for unit tests (that dont exist atm), the formal name for this pattern seems to be "dependency injection via constructor".

Going back to using a shared_ptr seems like it might solve most of the possible problems? Either way I'm not too worried about being 100% correct as long as it works, but it is an intresting point of conversation.

Reply:
It's generally bad practice to have (even non-const) references as member variables, as that article explains, because references cannot be rebound, i.e. are effectively const. There is no need to change back to `shared_ptr`, if you want you can use `std::reference_wrapper` instead, but I left this out of the review because I think the codebase is too simple for this to become a problem.